hoomd/hpmc/ComputeSDF.h

```diff
@@ -364,13 +384,181 @@ template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
             }     // end loop over images
-
-        // record the minimum bin
-        if ((unsigned int)min_bin < m_hist.size())
+        if (min_bin < m_hist.size())
+            {
             m_hist[min_bin]++;
+            }
+        } // end loop over all particles
+    }     // end countHistogramBinarySearch()
+
+template<class Shape> void ComputeSDF<Shape>::countHistogramLinearSearch(uint64_t timestep)
+    {
+    // update the aabb tree
+    const hoomd::detail::AABBTree& aabb_tree = m_mc->buildAABBTree();
+
+    // update the image list
+    const std::vector<vec3<Scalar>>& image_list = m_mc->updateImageList();
+
+    Scalar extra_width
+        = m_xmax / (1 - m_xmax) * (m_mc->getMaxCoreDiameter() + m_mc->getPatchEnergy()->getRCut());
+
+    // access particle data and system box
+    ArrayHandle<Scalar4> h_postype(m_pdata->getPositions(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<Scalar4> h_orientation(m_pdata->getOrientationArray(),
+                                       access_location::host,
+                                       access_mode::read);
+    ArrayHandle<Scalar> h_diameter(m_pdata->getDiameters(),
+                                   access_location::host,
+                                   access_mode::read);
+    ArrayHandle<Scalar> h_charge(m_pdata->getCharges(), access_location::host, access_mode::read);
+    const std::vector<param_type, hoomd::detail::managed_allocator<param_type>>& params
+        = m_mc->getParams();
+
+    // loop through N particles
+    // At the top of this loop, we initialize min_bin to the size of the sdf histogram
+    // For each of particle i's neighbors, we find the scaling that produces the first overlap.
+    // For each neighbor, we do a brute force search from the scaling corresponding to bin 0
+    // up to the minimum bin that we've already found for particle i.
+    // Then we add to m_hist[min_bin] the negative Mayer-function corresponding to the type
+    // of overlap corresponding to particle i's first overlap.
+    for (unsigned int i = 0; i < m_pdata->getN(); i++)
+        {
+        size_t min_bin = m_hist.size();
+        double hist_weight_ptl_i = 2.0;
+
+        // read in the current position and orientation
+        Scalar4 postype_i = h_postype.data[i];
+        Scalar4 orientation_i = h_orientation.data[i];
+        int typ_i = __scalar_as_int(postype_i.w);
+        Shape shape_i(quat<Scalar>(orientation_i), params[__scalar_as_int(postype_i.w)]);
+        vec3<Scalar> pos_i = vec3<Scalar>(postype_i);
 
+        // construct the AABB around the particle's circumsphere
+        // pad with enough extra width so that when scaled by xmax, found particles might touch
+        hoomd::detail::AABB aabb_i_local(vec3<Scalar>(0, 0, 0),
+                                         shape_i.getCircumsphereDiameter() / Scalar(2)
+                                             + extra_width);
```

Comment:
This is not correct. We should have a real time discussion with a white board to determine the modifications needed.

Reply:
I implemented the `getMaxInteractionDiameter()` method we discussed. I _think_ it does what it should now, but a close second look would be good.

Reply:
You still need to account for the patch `r_cut` and additive cutoff in the query AABB.

From `IntegratorHPMCMono.h` (with `extra_width`) added.
```
            OverlapReal r_cut_patch = 0;
            if (m_patch)
                {
                r_cut_patch = static_cast<OverlapReal>(m_patch->getRCut()) +
                    static_cast<OverlapReal>(0.5) * static_cast<OverlapReal>(m_patch->getAdditiveCutoff(typ_i));
                }


            // subtract minimum AABB extent from search radius
            OverlapReal R_query = std::max(shape_i.getCircumsphereDiameter()/OverlapReal(2.0),
                r_cut_patch-getMinCoreDiameter()/(OverlapReal)2.0);
            hoomd::detail::AABB aabb_i_local = hoomd::detail::AABB(vec3<Scalar>(0,0,0),R_query + extra_width);
```